pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -255,6 +231,42 @@ public long getCommittedTxnCount() {
 
     @Override
     public CompletableFuture<Position> appendBufferToTxn(TxnID txnId, long sequenceId, ByteBuf buffer) {
+        buffer.retain();
```

Comment:
When will this buffer be released?

Reply:
I am confusing why it will be released. Could you talk a little bit about its life cycle to help us understand why we need this retain?